diff --git a/clutter/clutter-fixed.c b/clutter/clutter-fixed.c
index 3027a75..315fe32 100644
--- a/clutter/clutter-fixed.c
+++ b/clutter/clutter-fixed.c
@@ -251,8 +251,8 @@ param_fixed_init (GParamSpec *pspec)
 {
   ClutterParamSpecFixed *fspec = CLUTTER_PARAM_SPEC_FIXED (pspec);
 
-  fspec->minimum = COGL_FIXED_MIN;
-  fspec->maximum = COGL_FIXED_MAX;
+  fspec->minimum = CLUTTER_MAXFIXED;
+  fspec->maximum = CLUTTER_MINFIXED;
   fspec->default_value = 0;
 }
 
