diff --git a/clutter/clutter-texture.c b/clutter/clutter-texture.c
index fc5541b..0f1a77e 100644
--- a/clutter/clutter-texture.c
+++ b/clutter/clutter-texture.c
@@ -457,7 +457,7 @@ clutter_texture_set_fbo_projection (ClutterActor *self)
 
   /* Set up a projection matrix so that the actor will be projected as
      if it was drawn at its original location */
-  tan_angle = clutter_tani (CLUTTER_ANGLE_FROM_DEGX (perspective.fovy / 2));
+  tan_angle = clutter_tanx (perspective.fovy / 2);
   near_size = CLUTTER_FIXED_MUL (perspective.z_near, tan_angle);
 
   cogl_frustum (CLUTTER_FIXED_MUL (x_min, near_size),
