diff --git a/clutter/clutter-texture.c b/clutter/clutter-texture.c
index b0b46e4..8a306e9 100644
--- a/clutter/clutter-texture.c
+++ b/clutter/clutter-texture.c
@@ -474,7 +474,7 @@ clutter_texture_set_fbo_projection (ClutterActor *self)
 
   /* Set up a projection matrix so that the actor will be projected as
      if it was drawn at its original location */
-  tan_angle = tanf (COGL_ANGLE_FROM_DEGX (perspective.fovy / 2 * (G_PI/180.0)));
+  tan_angle = tanf ((perspective.fovy / 2) * (G_PI/180.0));
   near_size = CLUTTER_FIXED_MUL (perspective.z_near, tan_angle);
 
   cogl_frustum (CLUTTER_FIXED_MUL (tx_min, near_size),
