diff --git a/clutter/cogl/cogl.h.in b/clutter/cogl/cogl.h.in
index cc26f88..f37d54d 100644
--- a/clutter/cogl/cogl.h.in
+++ b/clutter/cogl/cogl.h.in
@@ -271,9 +271,9 @@ void            cogl_translate                (gint                x,
  * rotation.
  */
 void            cogl_rotatex                  (float        angle,
-                                               gint                x,
-                                               gint                y,
-                                               gint                z);
+                                               float        x,
+                                               float        y,
+                                               float        z);
 
 /**
  * cogl_rotate:
@@ -442,7 +442,7 @@ void            cogl_enable_backface_culling  (gboolean            setting);
  * comparing with the value in @ref. The default function is CGL_ALWAYS the
  * initial reference value is 1.0.
  */
-void            cogl_alpha_func               (COGLenum            func, 
+void            cogl_alpha_func               (COGLenum            func,
                                                float        ref);
 
 /**
