diff --git a/clutter/cogl/gles/cogl-gles2-wrapper.h b/clutter/cogl/gles/cogl-gles2-wrapper.h
index ad741be..561cb66 100644
--- a/clutter/cogl/gles/cogl-gles2-wrapper.h
+++ b/clutter/cogl/gles/cogl-gles2-wrapper.h
@@ -203,8 +203,6 @@ struct _CoglGles2WrapperShader
 void cogl_gles2_wrapper_init (CoglGles2Wrapper *wrapper);
 void cogl_gles2_wrapper_deinit (CoglGles2Wrapper *wrapper);
 
-void cogl_wrap_glClearColorx (GLclampx r, GLclampx g, GLclampx b, GLclampx a);
-
 void cogl_wrap_glPushMatrix ();
 void cogl_wrap_glPopMatrix ();
 void cogl_wrap_glMatrixMode (GLenum mode);
@@ -239,12 +237,12 @@ void cogl_wrap_glDisableClientState (GLenum array);
 
 void cogl_wrap_glAlphaFunc (GLenum func, GLclampf ref);
 
-void cogl_wrap_glColor4x (GLclampx r, GLclampx g, GLclampx b, GLclampx a);
+void cogl_wrap_glColor4f (GLclampf r, GLclampf g, GLclampf b, GLclampf a);
 
 void cogl_wrap_glClipPlanex (GLenum plane, GLfloat *equation);
 
 void cogl_wrap_glGetIntegerv (GLenum pname, GLint *params);
-void cogl_wrap_glGetFixedv (GLenum pname, GLfloat *params);
+void cogl_wrap_glGetFloatv (GLenum pname, GLfloat *params);
 
 void cogl_wrap_glFogx (GLenum pname, GLfloat param);
 void cogl_wrap_glFogxv (GLenum pname, const GLfloat *params);
@@ -273,35 +271,34 @@ void _cogl_gles2_clear_cache_for_program (CoglHandle program);
 /* If we're not using GL ES 2 then just use the GL functions
    directly */
 
-#define cogl_wrap_glClearColorx        glClearColorx
 #define cogl_wrap_glDrawArrays         glDrawArrays
 #define cogl_wrap_glDrawElements       glDrawElements
 #define cogl_wrap_glPushMatrix         glPushMatrix
 #define cogl_wrap_glPopMatrix          glPopMatrix
 #define cogl_wrap_glMatrixMode         glMatrixMode
 #define cogl_wrap_glLoadIdentity       glLoadIdentity
-#define cogl_wrap_glMultMatrixx        glMultMatrixx
-#define cogl_wrap_glFrustumx           glFrustumx
-#define cogl_wrap_glScalex             glScalex
-#define cogl_wrap_glTranslatex         glTranslatex
-#define cogl_wrap_glRotatex            glRotatex
-#define cogl_wrap_glOrthox             glOrthox
+#define cogl_wrap_glMultMatrixx        glMultMatrixf
+#define cogl_wrap_glFrustumx           glFrustumf
+#define cogl_wrap_glScalex             glScalef
+#define cogl_wrap_glTranslatex         glTranslatef
+#define cogl_wrap_glRotatex            glRotatef
+#define cogl_wrap_glOrthox             glOrthof
 #define cogl_wrap_glEnable             glEnable
 #define cogl_wrap_glDisable            glDisable
 #define cogl_wrap_glTexCoordPointer    glTexCoordPointer
 #define cogl_wrap_glVertexPointer      glVertexPointer
 #define cogl_wrap_glColorPointer       glColorPointer
 #define cogl_wrap_glNormalPointer      glNormalPointer
-#define cogl_wrap_glTexEnvx            glTexEnvx
+#define cogl_wrap_glTexEnvx            glTexEnvf
 #define cogl_wrap_glEnableClientState  glEnableClientState
 #define cogl_wrap_glDisableClientState glDisableClientState
 #define cogl_wrap_glAlphaFunc          glAlphaFunc
-#define cogl_wrap_glColor4x            glColor4x
-#define cogl_wrap_glClipPlanex         glClipPlanex
+#define cogl_wrap_glColor4f            glColor4f
+#define cogl_wrap_glClipPlanex         glClipPlanef
 #define cogl_wrap_glGetIntegerv        glGetIntegerv
-#define cogl_wrap_glGetFixedv          glGetFixedv
-#define cogl_wrap_glFogx               glFogx
-#define cogl_wrap_glFogxv              glFogxv
+#define cogl_wrap_glGetFloatv          glGetFloatv
+#define cogl_wrap_glFogx               glFogf
+#define cogl_wrap_glFogxv              glFogfv
 #define cogl_wrap_glTexParameteri      glTexParameteri
 
 /* The extra third parameter of the bind texture wrapper isn't needed
