diff --git a/clutter/clutter-actor.c b/clutter/clutter-actor.c
index cc56310..4c89c0f 100644
--- a/clutter/clutter-actor.c
+++ b/clutter/clutter-actor.c
@@ -781,11 +781,10 @@ clutter_actor_real_allocate (ClutterActor          *self,
 
 /* Transform point (x,y,z) by matrix */
 static void
-mtx_transform (ClutterFixed m[16],
-	       ClutterFixed *x, ClutterFixed *y, ClutterFixed *z,
-	       ClutterFixed *w)
+mtx_transform (float m[16],
+	       float *x, float *y, float *z, float *w)
 {
-    ClutterFixed _x, _y, _z, _w;
+    float _x, _y, _z, _w;
     _x = *x;
     _y = *y;
     _z = *z;
